packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/404.tsx

```diff
@@ -0,0 +1,41 @@
+import { sitecorePagePropsFactory } from 'lib/page-props-factory';
+import config from 'temp/config';
+import { GraphQLErrorPagesService, SitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
+import { SitecorePageProps } from 'lib/page-props';
+import NotFound from 'src/NotFound';
+import { componentFactory } from 'temp/componentFactory';
+import Layout from 'src/Layout';
+import { GetStaticProps } from 'next';
+
+const Custom404 = (props: SitecorePageProps): JSX.Element => {
+  if (!(props && props.layoutData)) {
+    return <NotFound />;
+  }
+
+  return (
+    <SitecoreContext componentFactory={componentFactory} layoutData={props.layoutData}>
+      <Layout layoutData={props.layoutData} headLinks={props.headLinks} />
+    </SitecoreContext>
+  );
+};
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const props = await sitecorePagePropsFactory.create(context);
```

Comment:
It's maybe a little strange that we are attempting to fetch layoutData (for /404 I assume?) in page props factory (normalMode plugin), and then override it below. An alternative would be to fetch error pages first, and then pass the notFoundPagePath into the sitecorePagePropsFactory.create (context.params.path). 

Not sure that's better or worse though :) Just an observation.

Reply:
Yes, you're right! I changed this places and use only `siteResolver`